dependency_manager/src/edm_tool/bazel.py

```diff
@@ -0,0 +1,55 @@
+"Bazel related functions for edm_tool."
+import sys
+
+import yaml
+
+
+def generate_deps(args):
+    "Parse the dependencies.yaml and print content of *.bzl file to stdout."
+    with open(args.dependencies_yaml, 'r', encoding='utf-8') as f:
+        deps = yaml.safe_load(f)
+    print("Buid files: ", args.build_file, file=sys.stderr)
+    if args.build_file:
+        build_files = dict((f.split(":")[1], f) for f in args.build_file)
+    else:
+        build_files = {}
+
+    print('load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")')
+    print(
+        'load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")'
+    )
+    print()
+    print()
+
+    print("def edm_deps():")
+
+    for name, desc in deps.items():
+        repo = desc['git']
+        tag = desc["git_tag"]
+        commit = "None"
+
+        # Check that tag is hexadecimal 40-character string
+        if len(tag) == 40 and all(c in "0123456789abcdef" for c in tag.lower()):
+            tag, commit = "None", f'"{tag}"'
+        else:
+            tag, commit = f'"{tag}"', "None"
```

Comment:
Not sure about any differences, but wouldn't be `tag` instead of `f'"{tag}"'` be sufficient?  Or are you relying on some implicit default conversion of `None` to string?

Reply:
I want to achieve that in one case, the generated file will be:
```
tag = None
commit = "abcdef"
```
So in one case, we should insert a string whose value is `None`, and in the second we insert a string that has quotes around the commit hash.

I didn't come up with a better approach to handle this.

Reply:
I see, I missed the additional quotes.  Instead doing this logic directly on the variables, I would leave them as plain python strings (which could also be `None`) and create a helper function like
```
def quote_or_none(value: Optional[str]) -> str:
  return "None" if value == None else f'"{value}"'
```
and then reuse this function in the print string like
```
print(f"""
    ...
    tag = {quote_or_none(tag)}
    ...
```
This reduces code/logic duplication and simplifies the main code (although it is not that complex).

Furthermore I would add the helper function
```
def is_git_sha1(tag: str) -> bool:
  return len(tag) == 40 and all(c in "0123456789abcdef" for c in tag.lower())
```
which let you use ternary like
```
tag, commit = None, tag if is_git_sha1(tag) else tag, None
```

But of course, this might be up to personal taste.  Still using small functions with well chosen names, increases readability

Reply:
I've refactored the code quite a bit, can you take a look?